### PR TITLE
Add EVM header to block

### PR DESCRIPTION
**Pull request by Bushstar**
Add the EVM block header to the coinbase.

### CODE REVIEW

File: src/miner.cpp [hunk 1]
```diff
@@ -243,6 +243,13 @@ std::unique_ptr<CBlockTemplate> BlockAssembler::CreateNewBlock(const CScript& sc
         addPackageTxs<ancestor_score>(nPackagesSelected, nDescendantsUpdated, nHeight, mnview, evmContext);
     }
 
+    // TODO Get failed TXs and try to restore to mempool
+    std::array<uint8_t, 20> dummyAddress{};
+    const auto rustHeader = evm_finalize(evmContext, false, pos::GetNextWorkRequired(pindexPrev, pblock->nTime, consensus), dummyAddress);
+
+    std::vector<uint8_t> evmHeader{};
+    evmHeader.resize(rustHeader.size());
+    std::copy(rustHeader.begin(), rustHeader.end(), evmHeader.begin());
```
[the remark this answers is not in the document]
Comment (Bushstar) [reply src/miner.cpp]: Updated.

File: src/miner.cpp [hunk 2]
```diff
@@ -318,6 +330,16 @@ std::unique_ptr<CBlockTemplate> BlockAssembler::CreateNewBlock(const CScript& sc
             coinbaseTx.vout[0].nValue = CalculateCoinbaseReward(blockReward, consensus.dist.masternode);
         }
 
+        if (nHeight >= consensus.NextNetworkUpgradeHeight && !evmHeader.empty()) {
```
Comment (prasannavl) [inline src/miner.cpp:333]: Let's use `IsEvmEnabled` and move the fork check into `IsEVMEnabled` for all uses

File: src/miner.cpp [hunk 3]
```diff
@@ -243,6 +243,18 @@ std::unique_ptr<CBlockTemplate> BlockAssembler::CreateNewBlock(const CScript& sc
         addPackageTxs<ancestor_score>(nPackagesSelected, nDescendantsUpdated, nHeight, mnview, evmContext);
     }
 
+    const CDataStructureV0 enabledKey{AttributeTypes::Param, ParamIDs::Feature, DFIPKeys::EVMEnabled};
+    const auto attributes = mnview.GetAttributes();
+    assert(attributes);
+
+    // TODO Get failed TXs and try to restore to mempool
+    std::vector<uint8_t> evmHeader{};
+    if (nHeight >= consensus.NextNetworkUpgradeHeight && attributes->GetValue(enabledKey, false)) {
```
Comment (prasannavl) [inline src/miner.cpp:252]: Abstract into `IsEvmEnabled`

**Comment by prasannavl** [comment]
Ignore comments for now to keep diff down. Leaving them to be resolved post-merge.

**Pull request merged**